src/main/java/com/icthh/xm/ms/configuration/web/rest/ConfigurationClientResource.java

```diff
@@ -146,12 +147,29 @@ public ResponseEntity<Void> refreshConfiguration(HttpServletRequest request) {
     @PostMapping(value = PROFILE + "/configs_map")
     @Timed
     @LoggingAspectConfig(resultDetails = false)
-    public ResponseEntity<Map<String, Configuration>> getConfigurationsByPaths(@RequestBody List<String> paths) {
+    public ResponseEntity<Map<String, Configuration>> getConfigurationsByPaths(@RequestBody List<String> paths,
+                                                                               @RequestParam(name = "fetchAll", required = false, defaultValue = "false") Boolean fetchAll) {
         List<String> nonNullPaths = Optional.ofNullable(paths).orElseGet(Collections::emptyList);
-        Map<String, Configuration> configurations = configurationService.findConfigurations(nonNullPaths);
+        Map<String, Configuration> configurations = configurationService.findConfigurations(nonNullPaths, fetchAll);
         return ResponseEntity.ok(configurations);
     }
 
+    @GetMapping(value = PROFILE + "/configs_hash")
+    @Timed
+    @LoggingAspectConfig(resultDetails = false)
+    public ResponseEntity<ConfigurationsHashSumDto> getConfigurationsHashSum() {
+        return ResponseEntity.ok(configurationService.findConfigurationsHashSum());
+    }
+
+    @PostMapping(value = PROFILE + "/configs_update", consumes = {APPLICATION_JSON_VALUE})
+    @Timed
+    @PreAuthorize("hasPermission({'request': #request}, 'CONFIG.CLIENT.UPDATE.FROM.LIST')")
```

Comment:
propose rename permission from `CONFIG.CLIENT.UPDATE.FROM.LIST` to `CONFIG.CLIENT.UPDATE_LIST`

Reply:
renamed